### PR TITLE
MotionController Override

### PR DESCRIPTION
**Pull request by ave4224**
Allow MotionControllers to be overridden with an instance method. When overridden, they only do cursory calculations, like feed-forward multiplication of the input.

### CODE REVIEW

File: custom/motioncontrollers/MotionController.java [hunk 1]
```diff
@@ -256,27 +261,57 @@ public void enable() {
...
+	 */
+	public void kill() {
+		disable();
+		alive = false;
+	}
+	
+	/**
+	 * Resuscitates a controller, allowing disable() and enable() to be called again.
+	 */
+	public void resuscitate() {
```
Comment (carturn) [inline custom/motioncontrollers/MotionController.java]: Is there a shorter/easier to spell verb?

File: subsystems/motor/SensorMotor.java [hunk 2]
```diff
@@ -11,59 +11,76 @@
 public abstract class SensorMotor extends Motor {
 	protected final MotionController motionController;
 	private boolean isMotionControlEnabled;
-
-	public SensorMotor(String name, boolean inverted, SpeedModifier speedModifier, MotionController motionController,
-		SpeedController... motors) {
+	
+	public SensorMotor(String name, boolean inverted, SpeedModifier speedModifier, MotionController motionController, SpeedController... motors) {
```
Comment (carturn) [inline subsystems/motor/SensorMotor.java]: Please update your formatter before merging.
The split in this line is correct under the new paradigm of _avoid sideways scrolling_

File: custom/motioncontrollers/MotionController.java (hunk 1 above)
Comment (c0d3rman) [inline custom/motioncontrollers/MotionController.java]: perhaps have a single `setOverride` method? or an `enableOverride` and `disableOverride`?

**Comment by ajnadel** [comment]
Of note, I changed the essence of this PR. Now, it doesn't disallow disable() and enable(), but rather it only multiplies F*input. If we disallowed enabling, we'd have to simultaneously start using set() on the motors directly, as all of the setSetpoint() methods would not work because disabling() [which happens once at the time of override] would cancel the MotionControllerTask and the changing setpoint would never be reflected on the motor.

File: custom/motioncontrollers/MotionController.java [hunk 3]
```diff
@@ -278,6 +286,47 @@ public boolean isEnabled() {
...
 
 	/**
+	 * Set whether or not the motion controller
+	 * is overridden.
+	 * 
+	 * @see #startOverriding()
+	 * @see #stopOverriding()
+	 */
+	public void setOverride(boolean override) {
+		overridden = override;
```
Comment (c0d3rman) [inline custom/motioncontrollers/MotionController.java]: Please use the convention `this.overridden = overridden`

**Pull request merged**